lib/array_intersection.rb

```diff
@@ -1,6 +1,26 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n + m) where n is the number of items of one array and m is the number of items in the other
+# Space complexity: O(n)
 def intersection(array1, array2)
-  raise NotImplementedError
+  if array1 == nil || array2 == nil
+    return []
+  end
+  if array1.length < array2.length
```

Comment:
It doesn't really matter which array you put into the hash table since you have to go through both.  So this if statement is a bit wasted.